bloom-housing/bloom
# feat: public lottery view, totals data

Related to merge of: https://github.com/bloom-housing/bloom/pull/4275

This PR addresses #4224


## Description

Adds in the total number of applicants and number of applicants per preference into the public site lottery results page.

I created a new table that stores the totals in a very similar way to how we store lottery positions. I had performance concerns over querying the lottery positions table every time a user needs to see their results page as it will have 1000s of rows after just one lottery is run.

There's also some misc. test coverage.

## How Can This Be Tested/Reviewed?

Submit an application to a listing and claim a preference. Run and release the lottery. Ensure that the total number of applications and number of applications that claimed that preference are correct.

Handling listings without preferences or a user who doesn't claim any is a separate ticket.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
